ptp/ptp4u/drain/check.go

```diff
@@ -0,0 +1,7 @@
+package drain
+
+// Check is a drain check interface
+type Check interface {
+	// Drain returns true if the service needs to be drained
+	Drain() bool
```

Comment:
Drain sounds more like imperative aka drain now. Maybe “Check() bool”? And the interface itself could be Drain :)